### PR TITLE
hwmv2: Added Board Support for Starfive VisionFive2 RISC-V Single Board Computer

### PR DESCRIPTION
**Pull request by pfarwsi**
Modified the patches according to HWMV2 as per #69359. These list of patches contain Board support files for running Zephyr on the Starfive VisionFive2 RISC-V Single Board Computer based on JH7110 SOC. It's a basic support, no drivers other than UART have been tested.

### CODE REVIEW

File: boards/starfive/visionfive2_jh7110/visionfive2_jh7110.dts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include "starfive/jh7110-visionfive-v2.dtsi"
+
+/ {
+	model = "StarFive VisionFive V2";
+	compatible = "starfive,visionfive-v2", "starfive,jh7110";
```
Comment (fkokosinski) [inline boards/starfive/visionfive2_jh7110/visionfive2_jh7110.dts:13]:
```suggestion
	compatible = "starfive,visionfive-v2";
```

File: boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+# Zephyr Kernel Configuration
+CONFIG_PRINTK=y
+CONFIG_XIP=n
+
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_NS16550=y
```
Comment (fkokosinski) [inline boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig:11]: Continuing the discussion from https://github.com/zephyrproject-rtos/zephyr/pull/69359#discussion_r1502495926.

The SiFive UART is not enabled for HiFive Unmatched. You're likely referencing the `CONFIG_UART_SIFIVE_PORT_n` options - note that these are additional symbols. The Kconfig symbol for the driver itself is `CONFIG_UART_SIFIVE`, and is enabled by the `DT_HAS_SIFIVE_UART0_ENABLED` dts-generated config.

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig (hunk 2 above)
Comment (kanakshilledar) [reply boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig]: Updated the file accordingly

**Comment by kanakshilledar** [comment]
@fkokosinski any update on this PR?

**Comment by fkokosinski** [comment]
> @fkokosinski any update on this PR?

Hey, there are some issues issues with the PR and the changes it introduces:
* you have merge commits in your PR
* your commits are not squashed properly, making bisecting problematic in the future
* the documentation fails to build with your changes
* it looks like the board is not covered by any of the areas of maintenance as defined in the `MAINTAINERS.yml` file
* it also looks like you're adding a commit with me as a co-author. Please do not do that.

Also please not that as per Zephyr Project guidelines, reviewers have 1 week to reply to the changes contributors make in their PRs etc. While it usually take less than that, sometimes (e.g. when there's a new release coming, or in general - big changes are introduced) it may take some days before reviewers can get back to your PR :)

**Comment by kanakshilledar** [comment]
Hi @fkokosinski!
Thanks for you reply

> you have merge commits in your PR

We removed all the merges and just took our commits and pushed them.  You can see our commits from `0c208eca19127a858f76381c409e30b4a963168f` commit hash. 

> your commits are not squashed properly, making bisecting problematic in the future

We feel that these re submitting the changes might fix those

> the documentation fails to build with your changes

We checked the build logs and it reported loss of connectivity. 

> it looks like the board is not covered by any of the areas of maintenance as defined in the `MAINTAINERS.yml` file

We are adding ourselves as the Maintainers for this board. Under the `RISCV` area.

> it also looks like you're adding a commit with me as a co-author. Please do not do that.

We are doing this PR as a team of 2 people. With the whole SoC and Devicetree support by @pfarwsi and board support by me.

**Pull request closed without merging**